Marlin/src/module/temperature.cpp

```diff
@@ -632,6 +632,25 @@ volatile bool Temperature::raw_temps_ready = false;
  * Class and Instance Methods
  */
 
+#if ANY(HAS_PID_HEATING, MPC_AUTOTUNE)
+
+  /**
+   * Run the minimal required activities during a tuning loop.
+   * TODO: Allow tuning routines to call idle() for more complete keepalive.
+   */
+  inline void tuning_manage_activity() {
+    // Run HAL idle tasks
+    hal.idletask();
+
+    // Run Controller Fan check (normally handled by manage_inactivity)
+    TERN_(USE_CONTROLLER_FAN, controllerFan.update());
+
+    // Run UI update
+    TERN(DWIN_CREALITY_LCD, dwinUpdate(), ui.update());
```

Comment:
Aha, there is a general UI update… thought so. We can still poke through `manage_inactivity` to see if any other relevant items would be useful here, and we can still explore whether it would be okey-dokey to call `idle()` during PID/MPC tuning without having any negative impact on the tuning process. I don't think that's ever been attempted.